explainaboard/analysis/analyses.py

```diff
@@ -177,6 +177,63 @@ def perform(
 
         return BucketAnalysisResult(self.feature, bucket_performances)
 
+@dataclass
+class ComboCountAnalysisResult(AnalysisResult):
+    features: tuple
+    combo_counts: dict[tuple, int] | None = None
+    _type: Optional[str] = None
+
+    def __post_init__(self):
+        self._type: str = self.__class__.__name__
+
+    def print(self):
+        get_logger('report').info('feature combos for '+', '.join(self.features))
+        get_logger('report').info('\t'.join(self.features + ('#',)))
+        for k in sorted(self.combo_counts.keys()):
+            get_logger('report').info('\t'.join(k + (str(self.combo_counts[k]),)))
+        get_logger('report').info('')
+
+    def json_repr(self):
+        sanitized_counts: dict[str,int] = {}
+        for k, v in self.combo_counts.items():
+        return {'features': '\t'.join(self.features), 'combo_counts': sanitized_counts}
+
+
+@dataclass
+class ComboCountAnalysis(Analysis):
+    """
+    A class used to count feature combinations (e.g. for confusion matrices)
+    Args:
+        features: the name of the feature to bucket
+    """
+
+    features: tuple
+    _type: Optional[str] = None
+
+    def __post_init__(self):
+        self._type: str = self.__class__.__name__
+
+    AnalysisCaseType = TypeVar('AnalysisCaseType')
+
+    def perform(
+            self,
+            cases: list[AnalysisCase],
+            metrics: list[Metric],
+            stats: list[MetricStats],
+            conf_value: float,
+    ) -> AnalysisResult | None:
+        if len(cases) == 0 or any((x not in cases[0].features) for x in self.features):
+            return None
+        combo_counts: dict[tuple, int] = {}
+        for case in cases:
```

Comment:
It would be nice if some docstrings could be added here.

Reply:
Thanks for the comment @pfliu-nlp ! I've resolved it.